### PR TITLE
Added integration in Shard Distributor with the generated handler

### PR DESCRIPTION
**Pull request by jakobht**
**What changed?**
Added integration of the shard distributor IDLs to the server

**Why?**
We need to integrate this so we can implement the endpoint

**How did you test it?**

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: service/sharddistributor/handler/handler.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	metricsClient metrics.Client,
+	matchingPeerResolver matching.PeerResolver,
+) Handler {
+	handler := &handlerImpl{
+		logger:               logger,
+		metricsClient:        metricsClient,
+		matchingPeerResolver: matchingPeerResolver,
+	}
+
+	// prevent us from trying to serve requests before matching engine is started and ready
```
Comment (Shaddoll) [inline service/sharddistributor/handler/handler.go:46]: what is the matching engine in this context?

File: service/sharddistributor/handler/handler.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	"github.com/uber/cadence/client/matching"
+	"github.com/uber/cadence/common/log"
+	"github.com/uber/cadence/common/metrics"
+	"github.com/uber/cadence/common/types"
+)
+
+func NewHandler(
+	logger log.Logger,
+	metricsClient metrics.Client,
+	matchingPeerResolver matching.PeerResolver,
```
Comment (Shaddoll) [inline service/sharddistributor/handler/handler.go:38]: do we need history peer resolver?

File: service/sharddistributor/handler/handler.go (hunk 1 above)
Comment (davidporter-id-au) [reply service/sharddistributor/handler/handler.go:46]: @jakobht  what do you think about adding an arch diagram sketch into the `docs/arch` folder to contextualize what you're thinking / be able to point those of us (such as myself) for whom context may have fallen out of our brains.

Adding the mapper/handler shell all seems all good, but it'd probably help with review

File: service/sharddistributor/handler/handler.go (hunk 2 above)
Comment (jakobht) [reply service/sharddistributor/handler/handler.go:39]: I was thinking to focus on matching first, but we will need it, so added it.

File: service/sharddistributor/handler/handler.go (hunk 1 above)
Comment (jakobht) [reply service/sharddistributor/handler/handler.go]: @Shaddoll Sorry bad copy paste fixed the comment. 

@davidporter-id-au I think thats a good idea, added a task to do that

**Review by dkrotx — APPROVED** [review]

**Pull request merged**